### PR TITLE
Python3 wsystests

### PR DESCRIPTION
**Pull request by willrogers**
I made changes so that `dls_ade` runs on Python 3 as well as Python 2. Sky then ensured that the system tests run as well.

Are we happy to merge these changes into master?

### CODE REVIEW

**Comment by willrogers** [comment]
One thing I now remember is:

* We need python-ldap 3.x for compatibility with Python 3
* It is not easy to install python-ldap 3.x on RHEL6

So it's not straight-forward to reconcile these two things.

File: system_testing/system_testing.py [hunk 1]
```diff
@@ -400,8 +400,8 @@ def call_script(self):
                                    stdin=stdin_pipe)
 
         self._std_out, self._std_err = process.communicate(self._input)
-        self._std_out = self._std_out.decode('utf-8')
-        self._std_err = self._std_err.decode('utf-8')
+        self._std_out = self._std_out
+        self._std_err = self._std_err
```
Comment (hir12111) [inline system_testing/system_testing.py]: I believe we can delete those lines, but keep in mind that `self._std_out` was unicode before and the tests might expect that, also I've seen the same pattern in  `system_testing/checkout_module/checkout_module_sys_test.py:104` and `145`, those might need to be changed too

**Comment by ulrikpedersen** [comment]
> One thing I now remember is:
>
>    * We need python-ldap 3.x for compatibility with Python 3
>    * It is not easy to install python-ldap 3.x on RHEL6
>
>So it's not straight-forward to reconcile these two things.

I don't expect we will attempt to support python3 on RHEL6 anyway... So we have  a choice: do we maintain a parallel python3 branch - or do we maintain a parallel RHEL6 branch? Of course only until we can finally ditch RHEL6 entirely...

**Comment by willrogers** [comment]
_All right_, this now seems to work on Python 2 _and_ Python 3 _and_ RHEL6 _and_ RHEL7 (Python3/RHEL6 is not necessary). I wouldn't be surprised if there are a few edge cases that need clearing up still.

**Comment by willrogers** [comment]
Are we happy to merge this one while @ulrikpedersen's away? I think we're going to need these changes (specifically for the Python 3 deployment), so perhaps we merge and then fix any problems that are subsequently found.

**Comment by GDYendell** [comment]
I would like to merge #74 first and then rebase this one, if that isn't too painful to do.

**Comment by GDYendell** [comment]
Done! This can now be rebased on master.

**Pull request merged**